lib/core/storage.ts

```diff
@@ -11,21 +11,52 @@ function toBinary(str: string): string {
 
 class LocalStorage {
   private passportKey: string;
+  private targetingV1Key: string;
   private targetingKey: string;
 
   constructor(private Config: OptableConfig) {
     const sfx = btoa(toBinary(`${this.Config.host}/${this.Config.site}`));
+    // Legacy targeting key
+    this.targetingV1Key = "OPTABLE_TGT_" + sfx;
+
     this.passportKey = "OPTABLE_PASS_" + sfx;
-    this.targetingKey = "OPTABLE_TGT_" + sfx;
+    this.targetingKey = "OPTABLE_V2_TGT_" + sfx;
   }
 
   getPassport(): string | null {
     return window.localStorage.getItem(this.passportKey);
   }
 
-  getTargeting(): TargetingKeyValues | null {
-    const kvs = window.localStorage.getItem(this.targetingKey);
-    return kvs ? (JSON.parse(kvs) as TargetingKeyValues) : null;
+  getV1Targeting(): TargetingResponse | null {
+    const raw = window.localStorage.getItem(this.targetingV1Key);
+    const parsed = raw ? JSON.parse(raw) : null;
+    if (!parsed) {
+      return null
+    }
+
+    const audiences = Object.entries(parsed).map(([keyspace, values]) => {
+      return {
+        provider: "optable.co" as const,
```

Comment:
Having this here makes sense for backward compatibility but as mentioned, additional providers should be allowed without changes.